### PR TITLE
Fix crash with nullable activity on Android platform

### PR DESCRIPTION
**Pull request by zeienko-vitalii**
**Description of PR**
Contains a fix of unexpected application's crash when plugin tries to set null activity in `FlutterLocation.setActivity()` method.

**Description of the issue**
Such cases may happen when app's _Application.kt (*.java)_ class implements `PluginRegistrantCallback` and `registerWith(registry: PluginRegistry)` is called from other plugins that require to be registered that way (ex. flutter_messaging). `PluginRegistry` doesn't contain reference to the hosting activity and within  `Location.registerWith(Registrar registrar)` there is a call of `getFusedLocationProviderClient(activity)` method with null activity that throws a `NullPointerException` and this leads to crash 
#308 

**Influenced platforms**: Android

### CODE REVIEW

File: android/.settings/org.eclipse.buildship.core.prefs [hunk 1]
```diff
@@ -1,2 +1,2 @@
 connection.project.dir=../example/android
-eclipse.preferences.version=1
+eclipse.preferences.version=1
```
Comment (creativepsyco) [inline android/.settings/org.eclipse.buildship.core.prefs]: plz revert change in this file.

Comment (zeienko-vitalii) [reply android/.settings/org.eclipse.buildship.core.prefs:2]: Will do

Comment (zeienko-vitalii) [reply android/.settings/org.eclipse.buildship.core.prefs]: Resolved

**Comment by Roopeshlr-bounce** [comment]
@Lyokone  was this PR merged since we are facing the same issue ??

**Pull request closed without merging**